cognite/seismic/protos/v1/seismic_service_messages.proto

```diff
@@ -226,3 +226,81 @@ message SegYSeismicResponse {
 message SearchFilesRequest {
     SearchSpec spec = 1;
 }
+
+message RegisterFileRequest {
+    Identifier survey = 1;                                          // [required] Either name or id of a survey previously registered with /RegisterSurvey
+    string path = 2;                                                // [required if file is not synthetic] Path including protocol, bucket and directory structure. Example: "gs://cognite-seismic-eu/samples/"
+    string name = 3;                                                // [required] Unique filename including extension. Example: "DN1302M03R16_MERGED_KPSDM_00-32_DEG_T.sgy". The name must be unique across buckets and can be used to identify this file in query requests
+    map<string, string> metadata = 4;                               // [optional]
```

Comment:
At some point we should define som limits on the metadata object size. ie "no more than 100 keys, max 10 KB of data"